pkg/gcp/resources.go

```diff
@@ -330,38 +392,63 @@ func (r *gcpInstance) fromResourceDecription(resourceDesc []byte) (*computepb.In
 }
 
 // GCP cluster resource handler
-type gcpGKE struct {
-	gcpResourceHandler[container.ClusterManagerClient]
+type clusterHandler struct {
+	GCPResourceHandler
+	client          *container.ClusterManagerClient
+	firewallsClient *compute.FirewallsClient
```

Comment:
Do you think this is better than just embedding the entire `*GCPClients`? A benefit I can see with embedding `*GCPClients` is if `clusterHandler` ever needs another client, there will be minimal code changes.

Reply:
I thought it was nice to separate the initiation of the clients into its own function (saves a bunch of error checking and forces client errors to happen early), so if we keep that, we will have to enumerate which clients it uses anyways. I am not opposed to embedding the full thing, but I think these are fairly stable once created.